### PR TITLE
[TT-10515] Temporal Storage: TLS Support

### PR DESCRIPTION
**Pull request by mativm02**
## Description

Adding TLS support to Redis v8 interface. Also, I'm adding a new CI job to run tests against a TLS based Redis instance.
## Related Issue

https://tyktech.atlassian.net/browse/TT-10515?atlOrigin=eyJpIjoiOTVmNDkyM2NiMDA3NGI1ZjhkZjg3MTM5ZTE1MzczNzYiLCJwIjoiaiJ9
## Motivation and Context

https://tyktech.atlassian.net/browse/TT-10515?atlOrigin=eyJpIjoiOTVmNDkyM2NiMDA3NGI1ZjhkZjg3MTM5ZTE1MzczNzYiLCJwIjoiaiJ9
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)
- [ ] Documentation updates or improvements.

## Checklist

- [ ] I have [reviewed the guidelines](./CONTRIBUTING.md) for contributing to this repository.
- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [ ] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
     - [ ] I have manually updated the README(s)/documentation accordingly.
     - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
    - [ ] `gofmt -s -w .`
    - [ ] `go vet ./...`

### CODE REVIEW

File: bin/redis_TLS/cert.pem
Comment (mativm02) [inline bin/redis_TLS/cert.pem:1]: These are self-signed certificates used to deploy the TLS Redis and also to run unit tests using them. Are we okay leaving this here?

File: temporal/connector/connector_test.go [hunk 1]
```diff
@@ -1,97 +1,102 @@
...
-				MaxRetries: 3,
-			})},
-			expectedErr: nil,
-		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			connector, err := NewConnector(tt.typ, tt.opts...)
-			assert.Equal(t, tt.expectedErr, err)
+// func TestNewConnector(t *testing.T) {
```
Comment (tbuchaillot) [inline temporal/connector/connector_test.go:14]: @mativm02 commented

File: temporal/internal/tlsconfig/tlsconfig_test.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+		name        string
+		cfg         *model.TLS
+		expectedErr error
+	}{
+		{
+			name: "Valid config with Cert and Key",
+			cfg: &model.TLS{
+				Enable:             true,
+				InsecureSkipVerify: false,
+				CertFile:           "../../../bin/redis_TLS/cert.pem",
```
Comment (tbuchaillot) [inline temporal/internal/tlsconfig/tlsconfig_test.go:23]: Shouldn't we fetch this with env vars?

File: temporal/keyvalue/keyvalue_test.go [hunk 3]
```diff
@@ -1046,7 +1046,7 @@ func TestKeyValue_GetKeysWithOpts(t *testing.T) {
 			},
 			searchStr: "cursorkey*",
 			cursor:    0,
-			count:     20,
+			count:     50,
```
Comment (tbuchaillot) [inline temporal/keyvalue/keyvalue_test.go:1049]: Why this change 👀 ?

Comment (mativm02) [reply temporal/keyvalue/keyvalue_test.go:1049]: @tbuchaillot because the COUNT parameter in SCAN doesn't assure that X number of items will/won't be returned. From docs:
> While SCAN does not provide guarantees about the number of elements returned at every iteration, it is possible to empirically adjust the behavior of SCAN using the COUNT option. Basically with COUNT, the user specifies the amount of work that should be done at every call to retrieve elements from the collection. 

By setting a high value, I'm ensuring that most likely all of the 15 set values will be returned

File: bin/redis_TLS/cert.pem
Comment (mativm02) [reply bin/redis_TLS/cert.pem:1]: After a call with DevOps, we agreed to leave this as it is right now.

**Review by tbuchaillot — APPROVED** [review]
